### PR TITLE
Java: Add modelling for Apache HTTP Components

### PR DESCRIPTION
**Pull request by joefarebrother**

### CODE REVIEW

File: java/ql/src/semmle/code/java/frameworks/ApacheHttp.qll [hunk 1]
```diff
@@ -39,3 +40,242 @@ class TypeApacheHttpRequestBuilder extends Class {
     this.hasQualifiedName("org.apache.http.client.methods", "RequestBuilder")
   }
 }
+
+// TODO: Other sources
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/frameworks/ApacheHttp.qll:44]: Let's remove this comment. It's better to record remaining work in an issue.

**Review by aschackmull — COMMENTED** [review]
Otherwise LGTM. Do you prefer to merge this in its current form or do you want to use this as a testing ground for csv-based modelling? - it seems like most of this should convert fairly easily.

**Comment by joefarebrother** [comment]
I think I'll try converting it to the CSV modelling system

**Comment by joefarebrother** [comment]
@aschackmull I have converted it over.

**Review by aschackmull — APPROVED** [review]

**Pull request merged**